File: 1-two-sum/1-two-sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
 public:
  std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> num_to_compliment_idx;
    for (int i = 0; i < nums.size(); i++) {
      if (num_to_compliment_idx.count(nums[i])) {
        return {num_to_compliment_idx[nums[i]], i};
      }
      num_to_compliment_idx[target - nums[i]] = i;
    }
    return {-1, -1};
  }
};

File: 1-two-sum/1-two-sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numToComplimentIdx = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (numToComplimentIdx.containsKey(nums[i])) {
                return new int[] {numToComplimentIdx.get(nums[i]), i};
            }   
            numToComplimentIdx.put(target - nums[i], i);
        }
        return new int[]{-1, -1};
    }
}